packages/turbo/src/VisitableView.tsx

```diff
@@ -125,21 +103,32 @@ const VisitableView = React.forwardRef<RefObject, React.PropsWithRef<Props>>(
       [initializeStradaBridge, onLoad]
     );
 
+    const handleWarning = useCallback(
+      (e) => console.warn(e.nativeEvent.message),
+      []
+    );
+
     return (
       <>
         {stradaComponents?.map((Component, i) => (
           <Component
             key={i}
+            url={url}
             sessionHandle={sessionHandle}
             name={Component.componentName}
-            url={url}
+            registerMessageListener={registerMessageListener}
+            sendToBridge={sendToBridge}
```

Comment:
I'm thinking this could be improved.

As discussed, I am hugely in favor of exposing `VisitableView#injectJavaScript` to run some JS in the WebView, and an `onMessage` prop which will be called whenever there's a message _from_ the WebView.

The reason is that this maps 1:1 with the `react-native-webview` props and how other RN components would typically do this (`onSomething`) instead of an array of listeners. `injectJavaScript` maps to `WKWebView` method as well.

Reply:
A Strada HOC would simply provide an `onMessage` method, which would call the original if the message doesn't look like a Strada message:

```jsx
const withStrada =(Component) => ({ stradaComponents, onMessage, ...props }) => {
  const onMessageWithStrada = (message) => {
     // Do it's stuff
     
    onMessage?(message) // bubble to original prop
  }  

  return (
    <>
      .. strada components here
      <Component onMessage={onMessageWithStrada} {...props} />
    </>
  )
}

Reply:
> As discussed, I am hugely in favor of exposing VisitableView#injectJavaScript to run some JS in the WebView, and an onMessage prop which will be called whenever there's a message from the WebView.

All messages from native WebView are being passed to `onMessage` prop on `VisitableView`.
We expose VisitableView#injectJavaScript in its `ref`:
```jsx
    useImperativeHandle(
      ref,
      () => ({
        injectJavaScript: (callbackStringified) => {
          dispatchCommand(
            visitableViewRef,
            'injectJavaScript',
            callbackStringified
          );
        },
      }),
      []
    );
```

Reply:
Awesome! What do we need `registerMessageListener` and `sendToBridge` in that case? Wouldn't `onMessage` and `injectJavaScript` be enough for us?

Reply:
Apologies, I see that the `registerMessageListener` and `sendToBridge` are on the `BridgeComponent`, not the `VisitableView` :-)